### PR TITLE
br: fix bug in PiTR

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/34122

Problem Summary:

### What is changed and how it works?
- Remove checkpointTs and error information in pd when `br stop stream task`.
- Don't show checksum when `br start stream task`.
- Don't need to fix index when restore point, because it can backup and restore row KV and index KV.
- Hidden `filter` flag in `br restore point`, because it need support filtering table in current version.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/stream.go [hunk 1]
```diff
@@ -1047,15 +1045,14 @@ func restoreStream(
...
-	if err != nil {
-		return errors.Annotate(err, "failed to fix index for some table")
-	}
+	// pi := g.StartProgress(ctx, "Restore Index", countIndices(fullBackupTables), !cfg.LogProgress)
+	// err = withProgress(pi, func(p glue.Progress) error {
+	// 	return client.FixIndicesOfTables(ctx, fullBackupTables, p.Inc)
+	// })
+	// if err != nil {
+	// 	return errors.Annotate(err, "failed to fix index for some table")
+	// }
```
Comment (YuJuncen) [inline br/pkg/task/stream.go:1055]: Maybe not comment but just remove those code.

**Review by YuJuncen — APPROVED** [review]
rest lgtm

File: br/pkg/task/stream.go (hunk 1 above)
Comment (3pointer) [reply br/pkg/task/stream.go:1055]: +1

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**